Core/src/org/sleuthkit/autopsy/communications/VisualizationPanel.java

```diff
@@ -648,6 +681,14 @@ private void clearVizButtonActionPerformed(ActionEvent evt) {//GEN-FIRST:event_c
         setCursor(Cursor.getDefaultCursor());
     }//GEN-LAST:event_clearVizButtonActionPerformed
 
+    private void snapshotButtonActionPerformed(ActionEvent evt) {//GEN-FIRST:event_snapshotButtonActionPerformed
+        try{
+            handleSnapshotEvent();
```

Comment:
I would inline handleSnapshotEvent(), or alternatively break it and the related code out into another more focussed class.